lib/helper.js

```diff
@@ -71,5 +71,18 @@ module.exports = {
       }
     }
     return null;
+  },
+  /**
+   * Find the original caller that generate the error.
+   * @returns {*}
+   */
+  getServiceNameFromStackTrace: function() {
+    const stack = new Error().stack.toString();
+    const stackArr = stack.split(' at ');
+    const callers = stackArr.filter(trace => {
+      return /node-sdk/.test(trace) && !/node_modules/.test(trace) && !/anonymous/.test(trace) && trace;
+    });
+
+    return callers[callers.length - 1].split(' (')[0];
```

Comment:
@nfriendly you completely right, i think the refactor as you talked is the best solution. It's a bigger changes as i thought. All other easier solutions are non standard or obsolete, so i think the best is waiting for refactor and close this PR, what do you think ?